qsimcirq/qsim_simulator.py

```diff
@@ -22,6 +22,35 @@
 import qsimcirq.qsim_circuit as qsimc
 
 
+class QSimSimulatorState():
+
+    def __init__(self,
+                 qsim_data: np.ndarray,
+                 qubit_map: Dict[ops.Qid, int]):
+      self.qubit_map = qubit_map
+      # Generate state vector from qsim results
+      amplitudes = qsim_data
+      N = len(self.qubit_map)
+      amplitudes = np.reshape(amplitudes, [2]*(N+1))
+      amplitudes = np.transpose(amplitudes)
+      amplitudes = amplitudes[0] + 1j * amplitudes[1]
+      self.state_vector = amplitudes.flatten()
+
+
+class QSimSimulatorTrialResult(SimulationTrialResult):
```

Comment:
I think you want to inherit from WaveFunctionTrialResult here. It's what cirq.Simulator() returns and it takes care of setting the state_vector, plus some other convenience methods.

Reply:
Good point. Did similarly for the `QSimSimulatorState` object.